skbio/alignment/_path.py

```diff
@@ -0,0 +1,308 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import numpy as np
+
+from skbio._base import SkbioObject
+from skbio.util._decorator import overrides, classonlymethod
+
+
+class AlignPath(SkbioObject):
+    def __init__(self, lengths, states, starts):
+        """Create an alignment path from segment lengths and states.
+
+        Parameters
+        ----------
+        lengths : array_like of int of shape (n_segments,)
+            Length of each segment in the alignment.
+        states : array_like of unit8 of shape (n_segments,) or (n_packs, n_segments)
+            Packed bits representing character (0) or gap (1) status per sequence per
+            segment in the alignment.
+        starts : array_like of int of shape (n_sequences,), optional
+            Start position (0-based) of each sequence in the alignment.
+
+        See Also
+        --------
+        skbio.sequence.Sequence
+        skbio.alignment.TabularMSA
+
+        """
+        # Number of sequences needs to be explicitly provided, because the packed bits
+        # does not contain this information. (It is merely in multiples of 8.)
+        self.lengths = np.asarray(lengths, dtype=int)
```

Comment:
```suggestion
        self.lengths = np.asarray(lengths, dtype=dtype)
```
I wonder if we should have a `dtype` flag to be passed in, and have `np.uint32` be the default. 
But again, I'm not sure if this data structure is sufficient for genome-scale alignments (it can't recombination right?) Otherwise, these alignments could get *huge* (i.e. trillion bp genomes seen in some plants)

Reply:
This might be a good point. It may not be a concern in Linux or MacOS, but in Windows (even 64-bit), `int` defaults to `np.int32`, the maximum of which is 2,147,483,647 (i.e., 2 Gbp), which is shorter than many genomes. Perhaps we can default it to `np.uint64`. I am not sure if we need to create a dtype flag. This may confuse non-technical users.